source/PlayerInfo.cpp

```diff
@@ -2639,6 +2639,111 @@ void PlayerInfo::RegisterDerivedConditions()
 		return retVal;
 	});
 
+	// The total number of ships the player has.
+	auto &&totalShipsProvider = conditions.GetProviderNamed("total ships");
+	totalShipsProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed())
+				++retVal;
+		return retVal;
+	});
+
+	// Conditions to determine what outfits the player owns, with various possible locations to check.
+	// The following condition checks all sources of outfits which are present with the player.
+	// If in orbit, this means checking all ships in-system for installed and in cargo outfits.
+	// If landed, this means checking all landed ships for installed outfits, the pooled cargo hold,
+	// and any planetary storage.
+	auto &&presentOutfitProvider = conditions.GetProviderPrefixed("outfit (present): ");
+	presentOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
```

Comment:
This (and several others) won't work, will it... `name` is the entire condition name, isn't it...

Reply:
And that is also a good point; the function should indeed strip off the beginning of the condition name.

Reply:
This is already done for other conditions as well: https://github.com/endless-sky/endless-sky/blob/c723da748e5dae7580cbaa690dd99caec6e29b63/source/PlayerInfo.cpp#L2668

Reply:
I'll make use of that, then.